src/preprocessor.py

```diff
@@ -820,19 +820,15 @@ def check_group_daterange(self, group_df: pd.DataFrame, case_dr,
         """
         date_col = "date_range"
         delimiters = ",.!?/&-:;@_'\\s+"
-        if not hasattr(group_df, 'start_time') or not hasattr(group_df, 'end_time'):
-            if hasattr(group_df, 'time_range'):
-                start_times = []
-                end_times = []
-                for tr in group_df['time_range'].values:
-                    tr = tr.split('-')
-                    start_times.append(tr[0])
-                    end_times.append(tr[1])
-                group_df['start_time'] = pd.Series(start_times)
-                group_df['end_time'] = pd.Series(end_times)
-            else:
-                raise AttributeError('Data catalog is missing attributes `start_time` and/or'
-                                     ' `end_time` and can not infer from `time_range`')
+        if hasattr(group_df, 'time_range'):
+            start_times = []
+            end_times = []
+            for tr in group_df['time_range'].values:
+                tr = tr.split('-')
+                start_times.append(tr[0])
+                end_times.append(tr[1])
+            group_df['start_time'] = pd.Series(start_times)
+            group_df['end_time'] = pd.Series(end_times)
```

Comment:
@jtmims I recommend raising an error if no time_range attribute is found with a note that it is a required catalog entry.

Reply:
Okay, I wasn't sure if you wanted to go ahead and try with start_time and end_time. I will implement the suggestion!